src/apps/bridge/aanderaaController.cpp

```diff
@@ -0,0 +1,233 @@
+                                aanderaa->abs_speed_std_cm_s.getStd(),
+                                aanderaa->direction_circ_mean_rad.getCircularMean(),
+                                aanderaa->direction_circ_std_rad.getCircularStd(),
+                                aanderaa->temp_mean_deg_c.getMean());
+        BRIDGE_LOG_PRINTN(log_buf, log_buflen);
+        // TODO transmit data to spotter once we define it. 
+        aanderaa->abs_speed_mean_cm_s.clear();
+        aanderaa->abs_speed_std_cm_s.clear();
+        aanderaa->direction_circ_mean_rad.clear();
+        aanderaa->direction_circ_std_rad.clear();
+        aanderaa->temp_mean_deg_c.clear();
+      }
+    }
+  }
+}
+
+/*!
+ * @brief Initialize the Aandera controller.
+ * This controller is responsible for detecting Aanderaa nodes and subscribing to them.
+ * It will also aggregate the data from the Aanderaa nodes and transmit it over the spotter_tx service.
+ */
+void aanderaControllerInit(BridgePowerController *power_controller) {
+  configASSERT(power_controller);
+  _ctx._bridge_power_controller = power_controller;
+  BaseType_t rval = xTaskCreate(runController, "Aandera Controller", 128 * 4, NULL,
+                                AANDERAA_CONTROLLER_TASK_PRIORITY, &_ctx._task_handle);
+  configASSERT(rval == pdTRUE);
+}
+
+static void sampleTimerCallback(TimerHandle_t timer) {
+  (void)timer;
+  xTaskNotifyGive(_ctx._task_handle);
+}
+
+static void runController(void *param) {
+  (void)param;
+  if (_ctx._initialized) {
+    configASSERT(false); // Should only be initialized once
+  }
+  _ctx._subbed_aanderaas = NULL;
+  _ctx._num_subbed_aanderaas = 0;
+  _ctx._sample_timer = xTimerCreate("AanderaaCtlTim", pdMS_TO_TICKS(SAMPLE_TIMER_MS), pdTRUE,
+                                    NULL, sampleTimerCallback);
+  configASSERT(_ctx._sample_timer);
+  configASSERT(xTimerStart(_ctx._sample_timer, 10) == pdTRUE);
+  _ctx._initialized = true;
+  while (true) {
+    ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
+    if(_ctx._bridge_power_controller->waitForSignal(true,pdMS_TO_TICKS(TOPO_TIMEOUT_MS))) {
+        size_t size_list = sizeof(_ctx._node_list);
+        printf("Sampling for Aanderaa Nodes\n");
+        uint32_t num_nodes = 0;
+        if (topology_sampler_get_node_list(_ctx._node_list, size_list, num_nodes, TOPO_TIMEOUT_MS)) {
+        for (size_t i = 0; i < num_nodes; i++) {
+            if(_ctx._node_list[i] != getNodeId()) {
+                if (!sys_info_service_request(_ctx._node_list[i], node_info_reply_cb,
+                                        NODE_INFO_TIMEOUT_MS)) {
+                    printf("Failed to send sys_info request to node %" PRIx64 "\n", _ctx._node_list[i]);
+                }
+            }
+        }
+        }
+    }
+  }
+}
+
+static void createAanderaaSub(uint64_t node_id) {
+  Aanderaa_t *new_sub = reinterpret_cast<Aanderaa_t *>(pvPortMalloc(sizeof(Aanderaa_t)));
+  new_sub = new (new_sub) Aanderaa_t();
```

Comment:
First time using a placement new constructor 😅. I think this is the proper use case though - We want to allocate a class (struct) on the heap and do the proper construction, but because we're using an RTOS, we can't use `new Aanderaa_t` because of our dependancy on `pvPortMalloc`.

Reply:
TIL!

Reply:
TIL too! 😄